.gitlab-ci.yml

```diff
@@ -0,0 +1,215 @@
+# Copyright 2018 Red Hat, Inc.
+# Author: Jan Pokorny <jpokorny@redhat.com>
+# Part of pacemaker project
+# SPDX-License-Identifier: FSFAP
+
+variables:
+  GIT_DEPTH: "16"
+
+stages:
+  - buildenv
+  - build
+  - test
+
+.fedora-rawhide: &fedora-rawhide-anchor
+  tags:
+    - docker
+  image: fedora:rawhide
+
+.buildenv: &buildenv-anchor
+  stage: buildenv
+  variables:
+    GIT_STRATEGY: none
+
+.compilation: &compilation-anchor
+  only:
+   - pushes
+  # only since GitLab 11.4
+  #  changes:
+  #    - "configure.ac"
+  #    - "**/*.[ch]"
+  #    - "**/*.xml"
+  #    - "**/*.xsl"
+  #    - "**/Makefile.am"
+  #    - ".gitlab-ci.yml"
+
+buildenv:fedora-rawhide:info:
+  # primarily to gather some info about the image itself/container settings
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  script:
+   - cat /proc/meminfo | grep Free
+   - df -H
+   - lsblk
+   - nproc
+   - ulimit -a
+   - dnf --version
+
+buildenv:fedora-rawhide:rpms:
+  # primarily to download RPMs, but alas, "download" plugin not baked into image
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  cache:
+    key: "fedora-rawhide-dnf"
+    paths:
+      - ".cache/fedora-rawhide/dnf/rawhide-*"
+  script:
+    - mkdir -p .cache/fedora-rawhide/dnf
+    - cd .cache/fedora-rawhide
+    - mv dnf/rawhide-* /var/cache/dnf 2>/dev/null || true
+    - dnf-custom() { timeout 192 dnf -y --noplugins --nodocs
+                                 --setopt=tsflags=test
+                                 --setopt=install_weak_deps=0
+                                 --setopt=keepcache=1
+                                 --setopt=autocheck_running_kernel=0
+                                 --disablerepo='*' --enablerepo=rawhide "$@"; };
+      dnf-custom install
+        autoconf automake pkgconf-pkg-config libtool libtool-ltdl-devel
+        make findutils
+        gcc clang compiler-rt
+        libuuid-devel libxslt-devel bzip2-devel
+        'pkgconfig(glib-2.0) >= 2.16.0' 'pkgconfig(libxml-2.0)'
+        'pkgconfig(libqb) >= 0.13' 'pkgconfig(libcpg)'
+        libxml libxslt
+        psmisc which
+        bzip2-libs glib2
+        gcovr
```

Comment:
It would simplify a bit to define global variables for the reused dnf options and package dependencies

Reply:
True.  Likely not doable for functions at least since `shellshock` days,
and I'd rather avoid `eval`s, so they will stay copy-paste.